platform.md

```diff
@@ -518,13 +576,20 @@ In addition to the outputs enumerated below, outputs produced by `extender` incl
 | `1-10`, `13-19` | Generic lifecycle errors            |
 | `100-109`       | Extension-specific lifecycle errors |
 
-- For each extension in `<group>`, if a Dockerfile exists in `<generated>/build/<buildpack-id>`, the lifecycle:
-  - MUST apply the Dockerfile to the build environment according to the process outlined in the [Image Extension Specification](image-extension.md).
-- The extended image MUST be an extension of the `build-image` in [`analyzed.toml`](#analyzedtoml-toml)
-- After all Dockerfiles are applied, the lifecycle:
-  - MUST proceed with the `build` phase using the provided `<gid>` and `<uid>`
+- For each extension in `<group>` in order, if a Dockerfile exists in `<generated>/<kind>/<buildpack-id>`, the lifecycle:
+  - SHALL apply the Dockerfile to the environment according to the process outlined in the [Image Extension Specification](image-extension.md).
```

Comment:
`SHALL` is always odd to me. Does this mean it's required, like "MUST"?

Reply:
or is it optional like "MAY"?

Reply:
I've had to look this up myself. It's required: 

[1](https://www.rfc-editor.org/rfc/rfc2119#section-1). MUST   This word, or the terms "REQUIRED" or "SHALL", mean that the definition is an absolute requirement of the specification.